src/modeler/AutoRest.Swagger.Tests/SwaggerModelerTests.cs

```diff
@@ -250,6 +250,32 @@ public void TestcodeModelWithInheritance()
         }
 
         [Fact]
+        public void TestcodeModelWithXmsDiscriminatorValue()
+        {
+            using (NewContext)
+            {
+                new Settings
+                {
+                    Namespace = "Test",
+                    Input = Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "swagger-x-ms-discriminator-value.json")
+                };
+                var modeler = new SwaggerModeler();
+                var codeModel = modeler.Build();
+
+                Assert.NotNull(codeModel);
+                Assert.Equal("Pet", codeModel.ModelTypes.First(m => m.Name == "Pet").Name);
+                Assert.Equal("Microsoft.Models.MSPet", codeModel.ModelTypes.First(m => m.Name == "Pet").Extensions["x-ms-discriminator-value"]);
```

Comment:
@lamzuel - this test verifies that the base model "pet" has the extension on it. 

I generated the insights client locally by modifying the swagger (Base model RuleDataSource had the x-ms-discriminator-value extension applied to it) and here is the updated discriminator in NodeJS.

```javascript
exports.discriminators = {
  'RuleCondition' : exports.RuleCondition,
  >>>>>>>> 'Microsoft.Azure.Management.Insights.Models.RuleDataSource' : exports.RuleDataSource,
  'RuleDataSource.Microsoft.Azure.Management.Insights.Models.RuleMetricDataSource' : exports.RuleMetricDataSource,
  'RuleDataSource.Microsoft.Azure.Management.Insights.Models.RuleManagementEventDataSource' : exports.RuleManagementEventDataSource,
  'RuleCondition.Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition' : exports.ThresholdRuleCondition,
  'RuleCondition.Microsoft.Azure.Management.Insights.Models.LocationThresholdRuleCondition' : exports.LocationThresholdRuleCondition,
  'RuleCondition.Microsoft.Azure.Management.Insights.Models.ManagementEventRuleCondition' : exports.ManagementEventRuleCondition,
  'RuleAction' : exports.RuleAction,
  'RuleAction.Microsoft.Azure.Management.Insights.Models.RuleEmailAction' : exports.RuleEmailAction,
  'RuleAction.Microsoft.Azure.Management.Insights.Models.RuleWebhookAction' : exports.RuleWebhookAction
};
```